source/ldap-setup.rst

```diff
@@ -0,0 +1,357 @@
+
+To create the roles, use the following command:
+
+.. code-block:: javascript
+
+   var admin = db.getSiblingDB("admin")
+   db.createRole(
+      {
+        role: "cn=testusers,dc=percona,dc=com", 
+        privileges: [], 
+        roles: [ "readWrite"]
+      }
+   )
+
+   db.createRole(
+      {
+        role: "cn=otherusers,dc=percona,dc=com", 
+        privileges: [], 
+        roles: [ "read"]
+      }
+   )
+
+|PSMDB| configuration
+---------------------
+
+Access without username transformation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This section assumes that users connect to |PSMDB| by providing their LDAP :abbr:`DN (Distinguished Name)` as the username.
+
+1. Edit the |PSMDB| configuration file (by default, ``/etc/mongod.conf``) and specify the following configuration:
+   
+   .. code-block:: yaml
+
+      security:
+        authorization: "enabled"
+        ldap:
+          servers: "ldap.example.com"
+          transportSecurity: tls
+          authz:
+             queryTemplate: "dc=percona,dc=com??sub?(&(objectClass=groupOfNames)(member={PROVIDED_USER}))"
+          
+      setParameter:
+        authenticationMechanisms: "PLAIN"
+
+   Replace ``ldap.example.com`` with the hostname of your LDAP server. In the LDAP query template, replace the domain controllers ``percona`` and ``com`` with those relevant to your organization.
+
+#. Restart the ``mongod`` service:
+   
+   .. code-block:: bash
+
+      $ sudo systemctl restart mongod
+
+#. Test the access to |PSMDB|:
+   
+   .. code-block:: bash
+
+      mongo -u "cn=alice,dc=percona,dc=com" -p "secretpwd" --authenticationDatabase '$external' --authenticationMechanism 'PLAIN'
+
+Access with username transformation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If users connect to |PSMDB| with usernames that are not LDAP :abbr:`DN (Distinguished Name)`, you need to transform these usernames to be accepted by the LDAP server. 
+
+The ``--ldapUserToDNMapping`` configuration parameter allows you to do this. You specify the match pattern as a regexp to capture a username. Next, specify how to transform it - either to use a substitution value or to query the LDAP server for a username.
+
+If you don't know what the substitution or LDAP query string should be, please consult with the LDAP administrators to figure this out. 
+
+Note that you can use only the ``query`` or the ``substitution`` stage, the combination of two is not allowed.
+
+.. tabs::
+  
+   .. tab:: Substitution
+
+      1. Edit the |PSMDB| configuration file (by default, ``/etc/mongod.conf``) and specify the ``userToDNMapping`` parameter:
+         
+         .. code-block:: yaml
+
+            security:
+              authorization: "enabled"
+              ldap:
+                servers: "ldap.example.com"
+                transportSecurity: tls
+                authz:
+                   queryTemplate: "dc=percona,dc=com??sub?(&(objectClass=groupOfNames)(member={PROVIDED_USER}))"
```

Comment:
Should be {USER} instead of {PROVIDED_USER}, adding explanation might also be useful
https://docs.mongodb.com/manual/reference/configuration-options/#mongodb-setting-security.ldap.authz.queryTemplate

{USER}
Substitutes the authenticated username, or the transformed username, into the LDAP query.
{PROVIDED_USER}
Substitutes the supplied username, i.e. before either authentication or LDAP transformation, into the LDAP query. (Available starting in version 4.2)